flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/DefaultJobMasterServiceProcessTest.java

```diff
@@ -63,11 +68,181 @@ public void testInitializationFailureCompletesResultFuture() {
         final RuntimeException originalCause = new RuntimeException("Init error");
         jobMasterServiceFuture.completeExceptionally(originalCause);
 
-        assertTrue(serviceProcess.getResultFuture().join().isInitializationFailure());
-        final Throwable initializationFailure =
-                serviceProcess.getResultFuture().join().getInitializationFailure();
-        assertThat(initializationFailure, containsCause(JobInitializationException.class));
-        assertThat(initializationFailure, containsCause(originalCause));
+        final JobManagerRunnerResult actualJobManagerResult =
+                serviceProcess.getResultFuture().join();
+        assertThat(actualJobManagerResult.isInitializationFailure()).isTrue();
+        final Throwable initializationFailure = actualJobManagerResult.getInitializationFailure();
+
+        assertThat(initializationFailure)
+                .satisfies(
+                        t -> {
+                            assertThat(
+                                            ExceptionUtils.findThrowable(
+                                                            t, JobInitializationException.class)
+                                                    .isPresent())
+                                    .isTrue();
+                        });
+        CauseTreeContainsAssert.assertThat(initializationFailure)
+                .containsCause(JobInitializationException.class);
+        CauseTreeContainsAssert.assertThat(initializationFailure).containsCause(originalCause);
+    }
+
+    private static class CauseTreeContainsAssert
```

Comment:
I migrated to assertj when adding the new tests. I would move the custom Assert implementations along the lines of `flink-test-utils-junit` in `org.apache.flink.core.testutils` along `FlinkMatchers` and deprecate the `FlinkMatchers` class, if you agree. That would enable others to do the same with other tests in the future